util/compression.h

```diff
@@ -1593,4 +1594,125 @@ class CompressionTypeRecord {
   CompressionType compression_type_;
 };
 
+class StreamingCompress {
```

Comment:
Some comments describing these classes would be good. Also mention that this is not thread safe.